doc/users/faq/howto_faq.rst

```diff
@@ -9,6 +9,80 @@ How-to
 .. contents::
    :backlinks: none
 
+
+_how-to-too-many-ticks
+
+Why do I have so many ticks, and/or why are they out of order?
+--------------------------------------------------------------
+
+Sometimes Matplotlib will unexpectedly plot with a tick for each data point,
+and/or the ticks will be out of numerical order. This is usually a sign that you
+have passed in a list of strings rather than a list or array of floats or
+datetime objects.  This will often happen when reading in a comma-delimited text
+file. Matplotlib treats lists of strings as "categorical" variables
+(:doc:`/gallery/lines_bars_and_markers/categorical_variables`), and by default
+puts one tick per "category", and plots them in the order in which they are
+supplied.
+
+In the example below, the upper row plots are plotted using strings for *x*;
+note that each string gets a tick, and they are in the order of the list passed
+to Matplotlib. In the lower row the data is converted to either floats or
+datetime64; note that the ticks are now ordered and spaced numerically.
+
+.. plot::
+    :include-source:
+    :align: center
+
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    fig, ax = plt.subplots(2, 2, constrained_layout=True, figsize=(6, 6))
+    x = ['1', '5', '2', '3']
+    y = [1, 4, 2, 3]
+    ax[0, 0].plot(x, y, 'd')
+    ax[0, 0].set_xlabel('Categories')
+    # convert to numbers:
+    x = np.asarray(x, dtype='float')
+    ax[1, 0].plot(x, y, 'd')
+    ax[1, 0].set_xlabel('Floats')
+
+    x = ['2021-10-01', '2021-11-02', '2021-12-03', '2021-10-04']
+    y = [0, 2, 3, 1]
+    ax[0, 1].plot(x, y, 'd')
+    ax[0, 1].tick_params(axis='x', labelrotation=45)
+    # convert to datetime64
+    x = np.asarray(x, dtype='datetime64[s]')
+    ax[1, 1].plot(x, y, 'd')
+    ax[1, 1].tick_params(axis='x', labelrotation=45)
+
+If *x* has 100 elements, all strings, then we would have 100 (unreadable)
+ticks:
+
+.. plot::
+    :include-source:
+    :align: center
+
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    fig, ax = plt.subplots(figsize=(6, 2.5))
+    x = [f'{xx}' for xx in np.arange(100)]
+    y = np.arange(100)
+    ax.plot(x, y)
+
+
+.. _howto-determine-artist-extent:
+
+Determine the extent of Artists on the Figure
+---------------------------------------------
+
+Sometimes we want to know the extent of an Artist.  Matplotlib `.Artist` objects
+have a method `.Artist.get_window_extent` that will usually return the extent of
+the artist in pixels.  However, some artists, in particular text, must be
+rendered at least once before their extent is known.  Matplotlib supplies
+`.Figure.draw_without_rendering`, which should be called before calling
+``get_window_extent``.
```

Comment:
I guess I largely know of texts.  I guess there may be some artists that are proportional to the figure or axes size?

Reply:
I feel that vague statements are rather confusing to the user. We should do either

- dig into this deeper and explicitly write out for which artists this is necessary. - This is some work.
- State that users should always call `draw_without_rendering` before using `get_window_extent()`. - I'm a bit uneasy with that because we're forcing additional code and overhead on the user also in cases that would work without it.

I suggest to move this to a separate PR to not block the ticks stuff.

Note also that #22122 now mentions `draw_without_rendering()` in the `Text.get_window_extent()` exception, so that discussing `draw_without_rendering()` here for that case is less important now. If the error happens, the user has the solution at hand.

Reply:
I don't think people are coming to a "How to" for the exact reason their artist extent is wrong - they just want to fix it.  I don't think we need to enumerate all the troublesome artists, or interactions with layout managers etc here - we just want to tell them to call a  fake draw and see if that helps.

Reply:
Ok.